customer-commons/src/test/java/no/unit/nva/customer/service/impl/DynamoDBCustomerServiceTest.java

```diff
@@ -0,0 +1,230 @@
+        environment = mock(Environment.class);
+        service = new DynamoDBCustomerService(
+                objectMapper,
+                db.getTable()
+        );
+    }
+
+    @Test
+    public void testConstructorThrowsNoExceptions() {
+        when(environment.readEnv(TABLE_NAME)).thenReturn(NVA_CUSTOMERS_TEST);
+        CustomerService serviceWithTableNameFromEnv = new DynamoDBCustomerService(client, objectMapper, environment);
+        assertNotNull(serviceWithTableNameFromEnv);
+    }
+
+    @Test
+    public void createNewCustomerReturnsTheCustomer() throws Exception {
+        Customer customer = getNewCustomer();
+        Customer createdCustomer = service.createCustomer(customer);
+
+        assertNotNull(createdCustomer.getIdentifier());
+        assertEquals(customer, createdCustomer);
+    }
+
+    @Test
+    public void updateExistingCustomerWithNewName() throws Exception {
+        String newName = "New name";
+        Customer customer = getNewCustomer();
+        Customer createdCustomer = service.createCustomer(customer);
+        assertNotEquals(newName, createdCustomer.getName());
+
+        createdCustomer.setName(newName);
+        Customer updatedCustomer = service.updateCustomer(createdCustomer.getIdentifier(), createdCustomer);
+        assertEquals(newName, updatedCustomer.getName());
+    }
+
+    @Test
+    public void updateExistingCustomerChangesModifiedDate() throws Exception {
+        Customer customer = getNewCustomer();
+        Customer createdCustomer = service.createCustomer(customer);
+
+        Customer updatedCustomer = service.updateCustomer(createdCustomer.getIdentifier(), createdCustomer);
+        assertNotEquals(customer.getModifiedDate(), updatedCustomer.getModifiedDate());
+    }
+
+    @Test
+    public void updateExistingCustomerPreservesCreatedDate() throws Exception {
+        Customer customer = getNewCustomer();
+        Customer createdCustomer = service.createCustomer(customer);
+
+        Customer updatedCustomer = service.updateCustomer(createdCustomer.getIdentifier(), createdCustomer);
+        assertEquals(customer.getCreatedDate(), updatedCustomer.getCreatedDate());
+    }
+
+    @Test
+    public void getExistingCustomerReturnsTheCustomer() throws Exception {
+        Customer customer = getNewCustomer();
+        Customer createdCustomer = service.createCustomer(customer);
+        Customer getCustomer = service.getCustomer(createdCustomer.getIdentifier());
+        assertEquals(createdCustomer, getCustomer);
+    }
+
+    @Test
+    public void getAllCustomersReturnsListOfCustomers() throws Exception {
+        // create three customers
+        service.createCustomer(getNewCustomer());
+        service.createCustomer(getNewCustomer());
+        service.createCustomer(getNewCustomer());
+
+        List<Customer> customers = service.getCustomers();
+        assertEquals(3, customers.size());
+    }
+
+    @Test
+    public void getCustomerNotFoundThrowsException() {
```

Comment:
Maybe nitpicking, but the test means that the expected behavior of the method is to always throw an Exception :)
No need to fix, but I have to go to read the code to understand the intention.